corehq/apps/celery/shared_task.py

```diff
@@ -22,6 +22,9 @@ def task(*args, **kwargs):
     """
     kwargs.setdefault('serializer', 'json')
 
+    if len(args) == 1 and callable(args[0]):
+        return shared_task(**kwargs)(args[0])
```

Comment:
The commit message "add: support for task decorater with no args" implies that `kwargs` must be empty.

I can imagine an admittedly weird, possibly invalid, scenario that would trigger this branch, but probably should not:

```py
@task(dont_decorate_this_callable, max_retries=3)
def do_something():
    ...
```

Suggested change:
```suggestion
    if len(args) == 1 and callable(args[0]) and not kwargs:
        return shared_task()(args[0])
```

Reply:
I don't think this should be an issue but I like the idea of safeguarding it as much as possible.